crates/cac_client/src/lib.rs

```diff
@@ -157,29 +155,20 @@ impl Client {
     pub fn get_full_config_state_with_filter(
         &self,
         query_data: Option<Map<String, Value>>,
+        prefix: Option<Vec<String>>,
     ) -> Result<Config, String> {
         let mut config = self.config.read().map(|c| c.clone()).map_err_to_string()?;
-        if let Some(mut query_map) = query_data {
-            if let Some(prefix) = query_map.get("prefix") {
-                let prefix_list: HashSet<&str> = prefix
-                    .as_str()
-                    .ok_or_else(|| {
-                        log::error!("Prefix is not a valid string.");
-                        "Prefix is not a valid string.".to_string()
-                    })
-                    .map_err_to_string()?
-                    .split(',')
-                    .collect();
-                config =
-                    filter_config_by_prefix(&config, &prefix_list).map_err_to_string()?;
-            }
-
-            query_map.remove("prefix");
+        if let Some(prefix_list) = prefix {
+            config = filter_config_by_prefix(&config, &HashSet::from_iter(prefix_list))
```

Comment:
`nitpick`
Can we take HashSet as function args instead of building it here from Vector?